source/common/chromium_url/url_canon.h

```diff
@@ -8,8 +8,8 @@
 #ifndef URL_URL_CANON_H_
 #define URL_URL_CANON_H_
 
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
```

Comment:
higher level question for this file: this is obviously not written for Envoy, but appears to be forked from Chromium, I'm assuming because depending on its upstream was not tractable. 

But rather than bringing it into our coding style, should we add a third_party directory where we hold forked sources, and do not apply our current clang-tidy standards or naming conventions?

There's at least a few other Envoy naming-style divergences in this file as well, but it doesn't seem to make sense to fix them.

In fact all of the changes suggested here might make it harder to manually pull from Chromium later on if we decide we need to.

Reply:
+1 can we exclude this code?

Reply:
FWIW, we're not far away from being able to use the GURL library directly for this thanks to https://github.com/envoyproxy/envoy/pull/10599. So, if we can easily skip, that could simplify.